packages/ses/README.md

```diff
@@ -597,54 +597,95 @@ const compartmentB = new Compartment(harden({
 compartmentB.evaluate(programB);
 ```
 
-Any agent can also hide their own stack by causing client callbacks to
-run in a subsequent microtask.
+With `ses`, guest programs are initially powerless.
+A host can explicitly share limited powers with guest programs
+and provide intentional communication channels between them.
+
+Host programs must maintain the `ses` boundary with care in what they present
+as endowments.
+A host program should take care not to share mutable state with guests,
+or distribute mutable state to multiple guests, such as an unfrozen object (use
+`harden`), direct read and write access to a collection, like a `Map` or `Set`,
+even if hardened.
+Furthermore, typed arrays are collections and cannot be hardened.
 
-Among the ways a host program can invalidate these claims, a host can provide
+For the purposes of sharing state, pseudo-random number generators (PRNG) like
+`Math.random()` are equivalent to read and write access to shared state, and
+any guest can use one to eavesdrop on other guests or the host that share one.
 
-- either directly or by introduction through the transitive closure over
-  properties, property descriptors, arguments or returned values of functions
-  or methods:
-  - any shared mutable object,
-  - any resolution timer,
-  - any undesired communication channel, for example a PRNG,
-  - any `SharedArrayBuffer` that is indeed shared,
-  - the `Atomics` namespace object.
-- any function from from one agent that calls a callback from another agent on
-  its own stack (reentrancy hazards), including property descriptor functions
-  and proxy traps.
+If a guest program needs a high resolution timer to function, the host should
+only invite one guest to a single operating system process and limit the
+activity of the host program in the same process.
 
-These responsibilities for hosts are not exhaustive.
+Hosts must avoid exposing `SharedArrayBuffer` to guests.
+Any two JavaScript programs sharing a `SharedArrayBuffer` can use the shared
+buffer to construct a high resolution timer.
+
+The `ses` shim does not in itself isolate the stack of guest programs, even
+when evaluated in separate compartment.
+This is relevant when program created objects are shared between guest
+programs.
+
+Any object endowed to a program (either through the per-compartment
+`globalThis`, function arguments or returned values) represents a potential
+risk, such as reentrancy, when it interacts with a guest object.
```

Comment:
This may need some wordsmithing. It's not clear what `it` references to here. It should be program but the original subject is the endowed object.